ast/interpolation_test.go

```diff
@@ -21,6 +21,128 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestInvalidInterpolations(t *testing.T) {
+	cases := []struct {
+		text  string
+		parts []Interpolation
+	}{
+		{
+			text: "${foo",
+			parts: []Interpolation{
+				{
+					Text: "",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertyName{Name: "foo"},
+						},
+					},
+				},
+			},
+		},
+		{
+			text: "${foo ",
+			parts: []Interpolation{
+				{
+					Text: "",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertyName{Name: "foo"},
+						},
+					},
+				},
+				{
+					Text: " ",
+				},
+			},
+		},
+		{
+			text: `${foo} ${["baz} bar`,
+			parts: []Interpolation{
+				{
+					Text: "",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertyName{Name: "foo"},
+						},
+					},
+				},
+				{
+					Text: " ",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertySubscript{Index: "baz} bar"},
+						},
+					},
+				},
+			},
+		},
+		{
+			text: `missing ${property[} subscript`,
+			parts: []Interpolation{
+				{
+					Text: "missing ",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertyName{Name: "property"},
+						},
+					},
+				},
+				{
+					Text: " subscript",
+				},
+			},
+		},
+		{
+			text: `${[bar].baz}`,
+			parts: []Interpolation{
+				{
+					Text: "",
+					Value: &PropertyAccess{
+						Accessors: []PropertyAccessor{
+							&PropertySubscript{Index: 0},
```

Comment:
Is the `bar` discarded in this case?

Reply:
Yes, it is. We could instead treat this as a badly-formed string subscript and stash it in there. wdyt?

Reply:
(moving forward with this approach for now--we can always change this in the future)

Reply:
I think that's a reasonable choice. I just wanted to make sure I understood the test correctly.